src/test/java/springboot/controller/BookControllerTest.java

```diff
@@ -0,0 +1,225 @@
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+import springboot.dto.book.BookDto;
+import springboot.dto.book.BookDtoWithoutCategoryIds;
+import springboot.dto.book.CreateBookRequestDto;
+import springboot.testutil.BookUtil;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+
+    private static final String INSERT_BOOKS_SCRIPT_PATH =
+            "database/books/insert-books-with-categories-to-db.sql";
+    private static final String INSERT_CATEGORIES_SCRIPT_PATH =
+            "database/categories/insert-categories-to-test-db.sql";
+    private static final String REMOVE_ALL_SCRIPT_PATH =
+            "database/delete-all-data.sql";
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext,
+                          @Autowired DataSource dataSource) {
+        teardown(dataSource);
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @BeforeEach
+    void setUp(@Autowired DataSource dataSource) throws SQLException {
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource(INSERT_CATEGORIES_SCRIPT_PATH));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource(INSERT_BOOKS_SCRIPT_PATH));
+        }
+    }
+
+    @WithMockUser(username = "admin", authorities = {"ADMIN"})
+    @Test
+    @DisplayName("Create a book with valid parameters")
+    void createBook_ValidData_Success() throws Exception {
+        CreateBookRequestDto requestDto = BookUtil.createBookRequestDto();
+        BookDto expected = BookUtil.createBookDto(BookUtil.getBook(requestDto));
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(result
+                .getResponse()
+                .getContentAsString(), BookDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual);
```

Comment:
add (for all places)
```suggestion
        assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
```